File: app/src/main/java/com/extremedesign/typeracer/activity/SplashActivity.java
package com.extremedesign.typeracer.activity;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;

import com.extremedesign.typeracer.DataRepository.RepositoryTypeRacer.RepositoryViewModel;
import com.extremedesign.typeracer.FirebaseRepo;
import com.extremedesign.typeracer.R;
import com.extremedesign.typeracer.model.User;


public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);
        RepositoryViewModel repositoryViewModel=ViewModelProviders.of(this).get(RepositoryViewModel.class);
                        if(repositoryViewModel.isUserLoggedIn()){
                            Intent i = new Intent(SplashActivity.this,MainActivity.class);
                            i.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
                            startActivity(i);
                        }
                        else{
                            Intent i = new Intent(SplashActivity.this,LoginActivity.class);
                            i.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
                            startActivity(i);
                        }
                    }



}

File: app/src/main/java/com/extremedesign/typeracer/activity/SplashActivity.kt
package com.extremedesign.typeracer.activity

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.extremedesign.typeracer.R
import com.extremedesign.typeracer.data_repository.repository_typeracer.RepositoryViewModel
import com.extremedesign.typeracer.persistance.PreferencesTypeRacer
import com.extremedesign.typeracer.utils.GlobalConstants.USER_KEY
import org.koin.android.ext.android.inject

class SplashActivity : AppCompatActivity() {
    private val preferences: PreferencesTypeRacer by inject()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)
        if (preferences.getBoolean(USER_KEY)) {
            val i = Intent(this@SplashActivity, MainActivity::class.java)
            i.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
            startActivity(i)
        } else {
            val i = Intent(this@SplashActivity, LoginActivity::class.java)
            i.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
            startActivity(i)
        }
    }
}